src/actionmanager.cpp

```diff
@@ -436,14 +437,21 @@ void ActionManager::updateRecentsMenu()
                 action->setIcon(QIcon::fromTheme(type.iconName(), QIcon::fromTheme(type.genericIconName())));
 #else
                 // set icons for mac/windows users
+                QFileInfo fileInfo(recent.filePath);
                 QFileIconProvider provider;
-                action->setIcon(provider.icon(QFileInfo(recent.filePath)));
+                QIcon icon = provider.icon(fileInfo);
+#ifdef Q_OS_MACOS
+                // Workaround for native menu slowness
```

Comment:
Are we sure that the caching is only valuable on macOS?

Reply:
I was not able to observe any similar performance issues in Windows nor Linux Mint. Populating the Open With menu takes 0 ms on both, and populating the Recent menu takes maybe 2 to 3 ms on both. The Linux version is currently going through a different codepath to fetch/set the icon (the `QIcon::fromTheme` stuff), but I tried the other codepath and got the same performance. In both Windows and Linux, performance remained the same whether or not I had them go through `getCacheableIcon`.